pkgs/by-name/na/nav/package.nix

```diff
@@ -0,0 +1,65 @@
+{
+  stdenv,
+  lib,
+  fetchzip,
+  nix-update-script,
+  autoPatchelfHook,
+  libxcrypt-legacy,
+}:
+
+let
+  arch = if stdenv.hostPlatform.isAarch64 then "aarch64" else "x86_64";
+  platform = "${arch}-unknown-linux-gnu";
+in
+stdenv.mkDerivation rec {
+  pname = "nav";
+  version = "1.2.1";
+
+  src = fetchzip {
+    url = "https://github.com/Jojo4GH/nav/releases/download/v${version}/nav-${platform}.tar.gz";
+    sha256 =
+      {
+        x86_64-linux = "sha256-ihn5wlagmujHlSfJpgojQNqa4NjLF1wk2pt8wHi60DY=";
+        aarch64-linux = "sha256-l3rKu3OU/TUUjmx3p06k9V5eN3ZDNcxbxObLqVQ2B7U=";
+      }
+      .${stdenv.hostPlatform.system} or (throw "unsupported system ${stdenv.hostPlatform.system}");
+  };
```

Comment:
Could this not be built from source, using, e.g. `gradle2nix` rather than downloading the release binary?

Reply:
While looking for `gradle2nix`, I could not find any documentation in this repository at first and thus tried to get it running with [gradle](https://nixos.org/manual/nixpkgs/unstable/#gradle). However, we could not get the build running (our latest config is shown below).

This is the reason, I looked into `gradle2nix` again. The repository is currently migrating to [V2](https://github.com/tadfisher/gradle2nix/tree/v2) with functionality that is not (yet?) available for nixpkgs. We are currently wondering how to achieve the build and would like to know whether to stick to the binaries or if we should use:

```nix
let
  gradle2nix = import (fetchTarball "https://github.com/tadfisher/gradle2nix/archive/v2.tar.gz")  {};
in
gradle2nix.buildGradlePackage { ... }
```
*(as described in the documentation for `gradle2nix`)*

Would love to know how to proceed or where to get further information.

<details>
<summary>Nix Config for gradle.fetchDeps</summary>

We tried many variations of this file with `gradleUpdateTask`, `gradleUpdateScript`, and `gradleFlags`, but all of them failed at the dependency download or building step. @Jojo4GH's theory was that the dependency resolution does not work for `common` architectures, failing at downloading `kotlin-stdlib:2.0.0` for `arm64`.

```nix
{
  stdenv,
  lib,
  fetchFromGitHub,
  gradle,
  makeWrapper,
  jre,
  nix-update-script,
}:

stdenv.mkDerivation (finalAttrs: rec {
  pname = "nav";
  version = "1.2.1";

  src = fetchFromGitHub {
    owner = "Jojo4GH";
    repo = "nav";
    rev = "v${version}";
    hash = "sha256-rS64lvS8lBpOsNq+abqqLrX5HJsPTdVx6Akjg9S/pyQ=";
  };

  nativeBuildInputs = [
    gradle
  ];

  mitmCache = gradle.fetchDeps {
    inherit (finalAttrs) pname;
    data = ./deps.json;
  };

  __darwinAllowLocalNetworking = true;

  installPhase = ''
    runHook preInstall

    # this was never reached...

    runHook postInstall
  '';

  passthru.updateScript = nix-update-script { };

  meta = {
    description = "The Interactive and Stylish Replacement for ls & cd";
    longDescription = ''
      To make use of nav, add the following lines to your configuration:
      `programs.bash.shellInit = "eval \"$(nav --init bash)\"";` and
      `programs.zsh.shellInit = "eval \"$(nav --init zsh)\"";`
    '';
    homepage = "https://github.com/Jojo4GH/nav";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [
      David-Kopczynski
      Jojo4GH
    ];
    platforms = lib.platforms.linux;
    sourceProvenance = with lib.sourceTypes; [
      fromSource
      binaryBytecode # for gradle dependencies
    ];
    mainProgram = "nav";
  };
})
```
</details>

Reply:
I've left an approving review as I don't see any real problem with packaging it as you have, and I wouldn't want to do anything to discourage someone with the commit bit merging it.

I'm curious though, so I'll have a try packaging with the `v2` gradle2nix, and if successful that can go as a separate PR once this is merge